scr/main/java/bittercode/constant/ResponseCode.java

```diff
@@ -0,0 +1,42 @@
+package com.bittercode.constant;
+
+import java.util.Arrays;
+import java.util.Optional;
+
+public enum ResponseCode {
+
+    SUCCESS(200, "OK"),
+    FAILURE(422, "Unprocessible Entity, Failed to Process"),
+    PAGE_NOT_FOUND(404, "The Page You are Searching For is Not available"),
+    ACCESS_DENIED(403, "Please Login First to continue"),
+    BAD_REQUEST(400, "Bad Request, Please Try Again"),
+    INTERNAL_SERVER_ERROR(500, "Internal Server Error, Try Again!!"),
+    DATABASE_CONNECTION_FAILURE(406,
+            "Unable to Connect to DB, Please Check your db credentials in application.properties"),
+    METHOD_NOT_ALLOWED(405, "Requested HTTP method is not supported by this URL"),
+
+    ;
+
+    private final String message;
+    private final int code;
+
+    ResponseCode(int code, String message) {
+        this.message = message;
+        this.code = code;
+    }
+
+    public String getMessage() {
+        return message;
+    }
+
+    public int getCode() {
+        return code;
+    }
+
+    // Reverse lookup methods
+    public static Optional<ResponseCode> getMessageByStatusCode(int statusCode) {
+        return Arrays.stream(ResponseCode.values())
+                .filter(error -> error.getCode() == statusCode)
+                .findFirst();
+    }
```

Comment:
<div>


<div id="suggestion">
<div id="issue"><b>Method name may not reflect functionality</b></div>
<div id="fix">

The method `getMessageByStatusCode` might be more appropriately named `getResponseCodeByStatusCode` as it returns a `ResponseCode` object, not just a message. Consider renaming for clarity.
</div>
<details>
<summary>
<b>Code suggestion</b>
</summary>
<div id="code">

```
 @@ -37,5 +37,5 @@
 -    public static Optional<ResponseCode> getMessageByStatusCode(int statusCode) {
 +    public static Optional<ResponseCode> getResponseCodeByStatusCode(int statusCode) {
          return Arrays.stream(ResponseCode.values())
                  .filter(error -> error.getCode() == statusCode)
                  .findFirst();
      }
```
</div>
</details>
</div>

<div id="suggestion">
<div id="issue"><b>Consider more efficient lookup for getMessageByStatusCode</b></div>
<div id="fix">

The `getMessageByStatusCode` method might benefit from using a more efficient lookup mechanism. Consider using a `Map` for constant-time lookups instead of streaming through all values.
</div>
<details>
<summary>
<b>Code suggestion</b>
</summary>
<div id="code">

```
 @@ -1,5 +1,7 @@
  package bittercode.constant;
 
  import java.util.Arrays;
  import java.util.Optional;
 +import java.util.Map;
 +import java.util.HashMap;
 
  public enum ResponseCode {
      // ... existing enum values ...
 +    private static final Map<Integer, ResponseCode> CODE_MAP = new HashMap<>();
 
      private final String message;
      private final int code;
 @@ -23,6 +25,12 @@
          this.message = message;
          this.code = code;
      }
 +
 +    static {
 +        for (ResponseCode rc : values()) {
 +            CODE_MAP.put(rc.getCode(), rc);
 +        }
 +    }
 
      public String getMessage() {
          return message;
 @@ -37,8 +45,7 @@
      // Reverse lookup methods
      public static Optional<ResponseCode> getMessageByStatusCode(int statusCode) {
 -        return Arrays.stream(ResponseCode.values())
 -                .filter(error -> error.getCode() == statusCode)
 -                .findFirst();
 +        return Optional.ofNullable(CODE_MAP.get(statusCode));
      }
  }
```
</div>
</details>
</div>


<small><i>Code Review Run <a href=https://github.com/PrajaktaBendreBito/BookStore/pull/1#issuecomment-2437361366>#e347cb</a></i></small>
</div>

---
Is this a valid issue, or was it incorrectly flagged by the Agent?
- [ ] it was incorrectly flagged